daemon/cmd/fqdn.go

```diff
@@ -336,11 +336,9 @@ func (d *Daemon) notifyOnDNSMsg(
 		defer updateCancel()
 		updateStart := time.Now()
 
-		dpUpdates := d.dnsNameManager.UpdateGenerateDNS(updateCtx, lookupTime, map[string]*fqdn.DNSIPRecords{
-			qname: {
-				IPs: responseIPs,
-				TTL: int(TTL),
-			},
+		dpUpdates := d.dnsNameManager.UpdateGenerateDNS(updateCtx, lookupTime, qname, &fqdn.DNSIPRecords{
```

Comment:
Nit on the commit name: this is avoiding an intermediate `map`, so the savings are easier to comprehend than would be a case of a `struct`.